### PR TITLE
Pub-Sub Auth Push sample

### PR DESCRIPTION
**Pull request by hemanshv**
Added pub-sub auth push [sample]([url](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/issues/1234))
Please review

### CODE REVIEW

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
I think you probably need to modify the kestrel tests to test this new method. There's a *.json with the config, but I'm not very familiar with it. Let m know if you need help with that.

@anguillanneuf I'll request your review mostly to discuss a little about caching the tokens. I'll also leave a note on the internal emails.

**Comment by hemanshv** [comment]
Addressed the review comments pls review

**Comment by amanda-tarafa** [comment]
This will most likely need to wait till I'm back. But also, the tests are failing.

**Comment by hemanshv** [comment]
@amanda-tarafa The local tests are resolved Please review once your are back, Thanks

**Comment by hemanshv** [comment]
@amanda-tarafa  Please review

File: appengine/flexible/Pubsub/Controllers/HomeController.cs [hunk 1]
```diff
@@ -85,11 +90,60 @@ public IActionResult Index(MessageForm messageForm)
...
+            // overhead if you verify the tokens offline by decoding them using
+            // Google's Public Cert; caching already seen tokens works best when
+            // a large volume of messages have prompted a single push server to
+            // handle them, in which case they would all share the same token for
+            // a limited time window.
+            var payload = await JsonWebSignature.VerifySignedTokenAsync<PubSubPayload>(authToken);
+
+            // IMPORTANT: you should validate payload details not covered
+            // by signature and audience verification above, including:
+            //   - Ensure that `payload.email` is equal to the expected service
```
Comment (amanda-tarafa) [inline appengine/flexible/Pubsub/Controllers/HomeController.cs:136]:
```suggestion
            //   - Ensure that `payload.Email` is equal to the expected service

```

**Review by amanda-tarafa — APPROVED** [review]
LGTM with 3 tiny nits!

Thanks!

**Comment by amanda-tarafa** [comment]
Thanks! I'll merge on green.

**Pull request merged**